pages/solana.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useEffect, useState } from "react";
+import styles from "../styles/solana.module.css";
+import Image from "next/image";
+import AffiliateImage from "../public/visuals/affiliate.webp";
+import Button from "../components/UI/button";
+import StarknetIcon from "../components/UI/iconsComponents/icons/starknetIcon";
+import Wallets from "../components/UI/wallets";
+import { useAccount, useContractWrite } from "@starknet-react/core";
+import ProgressBar from "../components/UI/progressBar";
+import { NextPage } from "next";
+import { WalletMultiButton } from "@solana/wallet-adapter-react-ui";
+import { useWallet } from "@solana/wallet-adapter-react";
+import { Call } from "starknet";
+import TxConfirmationModal from "../components/UI/txConfirmationModal";
+import { NotificationType, TransactionType } from "../utils/constants";
+import { useNotificationManager } from "../hooks/useNotificationManager";
+import { Skeleton } from "@mui/material";
+import SolanaCalls from "../utils/callData/solanaCalls";
+import { utils } from "starknetid.js";
+import useHasClaimSolSubdomain from "../hooks/useHasClaimSolSubdomain";
+import DoneFilledIcon from "../components/UI/iconsComponents/icons/doneFilledIcon";
+import theme from "../styles/theme";
+import DomainActions from "../components/solana/domainActions";
```

Comment:
Could we have an `open/close` state ? I don't want us to have to update bugs and solana wallet package versions in the future so I prefer we simply let people for 3 weeks then close and reopen in the future. 

What do you think ? You can reuse the close state of discounts we made in the past (last one is the freerenewal)